src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -230,6 +231,11 @@ class AdvertiserMinMdns : public ServiceAdvertiser,
         if (optionalMrp.HasValue())
         {
             auto mrp = optionalMrp.Value();
+
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+            // A ICD operating as a LIT should not advertise its slow polling interval
+            if (!params.GetICDOperatingAsLIT().HasValue() || !params.GetICDOperatingAsLIT().Value())
```

Comment:
It's worth documenting why the `!HasValue()` case should advertise things.  What does that case correspond to?

Reply:
When the ICD doesn't support LIT features, It doesn't advertise the ICD key at all and it has to provide the SII key

so it is basically
ICD SIT -> advertises SII key but no ICD KEY
ICD LIT but currently operating as SIT -> advertises SII key and ICD=0
ICD LIT and operating as such -> ICD=1 and no SII key

Will add this to the comments